server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -362,7 +369,7 @@ public Repository repository(String repositoryName) {
      * @param repositoryMetaData new repository metadata
      * @return {@code true} if new repository was added or {@code false} if it was ignored
      */
-    private boolean registerRepository(RepositoryMetaData repositoryMetaData) throws IOException {
+    private boolean registerRepository(RepositoryMetaData repositoryMetaData) {
         Repository previous = repositories.get(repositoryMetaData.name());
```

Comment:
I wonder if this logic is still needed. If we want to keep the logic to check if there have been no changes, we should just take the current repository metadata from the cluster state, and do this on the cluster state update task.

Reply:
you can then also inline this method so that it only becomes something like
```
try {
    closeRepository(createRepository(newRepositoryMetaData));
} catch (Exception e) {
    registrationListener.onFailure(e);
    return;
}
```

Reply:
Makes sense :) Inlined as you suggest above and added a breakout to the cluster state task for the case of equal metadata.